cmd/enterprise-portal/service/service.go

```diff
@@ -32,27 +43,94 @@ func (Service) Initialize(ctx context.Context, logger log.Logger, contract runti
 	if err != nil {
 		return nil, errors.Wrap(err, "newDotComDBConn")
 	}
-	defer dotcomDB.Close(context.Background())
 
 	// Validate connection on startup
-	if err := dotcomDB.Ping(context.Background()); err != nil {
+	if err := dotcomDB.Ping(ctx); err != nil {
 		return nil, errors.Wrap(err, "dotcomDB.Ping")
 	}
+	logger.Debug("connected to dotcom database")
+
+	// Prepare SAMS client, so that we can enforce SAMS-based M2M authz/authn
+	logger.Debug("using SAMS client",
+		log.String("samsExternalURL", config.SAMS.ExternalURL),
+		log.Stringp("samsAPIURL", config.SAMS.APIURL),
+		log.String("clientID", config.SAMS.ClientID))
+	samsClient, err := sams.NewClientV1(
+		sams.ClientV1Config{
+			ConnConfig: config.SAMS.ConnConfig,
+			TokenSource: sams.ClientCredentialsTokenSource(
+				config.SAMS.ConnConfig,
+				config.SAMS.ClientID,
+				config.SAMS.ClientSecret,
+				[]scopes.Scope{scopes.OpenID, scopes.Profile, scopes.Email},
+			),
+		},
+	)
+	if err != nil {
+		return nil, errors.Wrap(err, "create Sourcegraph Accounts client")
+	}
 
 	httpServer := http.NewServeMux()
 
 	// Register MSP endpoints
 	contract.Diagnostics.RegisterDiagnosticsHandlers(httpServer, serviceState{})
 
-	// Initialize server
+	// Register connect endpoints
+	codyaccessservice.RegisterV1(logger, httpServer, samsClient.Tokens(), dotcomDB)
+	subscriptionsservice.RegisterV1(logger, httpServer)
+
 	listenAddr := fmt.Sprintf(":%d", contract.Port)
+	if !contract.MSP && debugserver.GRPCWebUIEnabled {
+		// Enable reflection for the web UI
+		reflector := grpcreflect.NewStaticReflector(
+			codyaccessservice.Name,
+			subscriptionsservice.Name,
+		)
+		httpServer.Handle(grpcreflect.NewHandlerV1(reflector))
+		httpServer.Handle(grpcreflect.NewHandlerV1Alpha(reflector)) // web UI still requires old API
+		// Enable the web UI
+		grpcUI := debugserver.NewGRPCWebUIEndpoint("enterprise-portal", listenAddr)
+		httpServer.Handle(grpcUI.Path, grpcUI.Handler)
+		logger.Warn("gRPC web UI enabled", log.String("url", fmt.Sprintf("%s%s", listenAddr, grpcUI.Path)))
+	}
+
+	// Initialize server
 	server := httpserver.NewFromAddr(
 		listenAddr,
 		&http.Server{
-			ReadTimeout:  2 * time.Minute,
-			WriteTimeout: 2 * time.Minute,
-			Handler:      httpServer,
+			Addr: listenAddr,
+			Handler: h2c.NewHandler(
+				otelhttp.NewHandler(
+					// Cloud Run only supports HTTP/2 if the service accepts HTTP/2 cleartext (h2c),
+					// see https://cloud.google.com/run/docs/configuring/http2
```

Comment:
```suggestion
			// Cloud Run only supports HTTP/2 if the service accepts HTTP/2 cleartext (h2c),
			// see https://cloud.google.com/run/docs/configuring/http2
			Handler: h2c.NewHandler(
				otelhttp.NewHandler(
```

Meant to fix this in SAMS but keep forgetting 😂 the docstring has been out of place ever since we swapped the wrapping.

Reply:
Ah, thank! I just copy-pasted rather blindly from SAMS 😆

Reply:
😂 That's exactly how people know code is copied